scipy/stats/tests/test_stats.py

```diff
@@ -2419,7 +2539,8 @@ def test_ttest_1samp_new():
 
         # check that nan in input array result in nan output
         anan = np.array([[1,np.nan],[-1,1]])
-        assert_equal(stats.ttest_1samp(anan, 0),([0, np.nan], [1,np.nan]))
+        # assert_equal(stats.ttest_1samp(anan, 0),([0, np.nan], [1,np.nan]))
+        assert_equal(stats.ttest_1samp(anan, 0),(np.nan, np.nan))
```

Comment:
nan input test was modified. Please review to make sure it won't cause problems. Like backwards compat break.